lumicks/pylake/channel.py

```diff
@@ -504,6 +504,137 @@ def plot(self, start=None, **kwargs):
+        ----------
+        item : Marker | tuple(int, int) | int | str
+            An item with a start and stop property such as a `Marker` or `CalibrationItem`, a
+            tuple of timestamps, a single timestamp or a string in time format (e.g. "5s" for
+            5 seconds).
+        annotation : str | None
+            String to annotate with
+        annotation_direction : "horizontal" | "vertical" | None
+            Which direction to show the text when annotating.
+        color : matplotlib.color | None
+            Matplotlib color
+        label : str | None
+            Label to use in legend entry if a legend is plotted
+        start : int | None
+            Plot with respect to a different "zero" timestamp.
+        alpha : float
+            Opacity (0 is transparent, 1.0 is opaque). Default: 0.5.
+        **kwargs :
+            Arguments forwarded to `matplotlib.axvline` and `matplotlib.axvspan`.
+
+        Examples
+        --------
+        ::
+
+            from lumicks import pylake
+
+            f = lk.File("data.h5")
+
+            plt.figure()
+            slc = ["Distance"]["Distance 1"]
+            slc.plot()
+
+            # Draws the time region spanned by marker 1
+            slc.highlight_time_range(f.markers["1"])
+
+            # Draws an annotation at the 1-second mark
+            slc.highlight_time_range("1s", annotation="first second")
+
+            # Draws a highlighted region from 1 to 3 seconds and marks it with the text "region"
+            slc.highlight_time_range(("1s", "3s"), annotation="region")
+
+            # Draws a highlighted region from 4 to 5 seconds, marks it, but forces text vertically
+            slc.highlight_time_range(("1s", "3s"), annotation="region", annotation_direction="vertical")
+
+
+            plt.figure()
+            f.force1x.plot()
+
+            item = f.force1x.calibration[1]
+
+            # Fills the region between calibration_item.start and calibration_item.stop
+            f.force1x.highlight_time_range(item, annotation="calibration 1")
+
+            # Draws a line with annotation at the time the calibration was applied.
+            f.force1x.highlight_time_range(item.applied_at, annotation="calibration 1 applied")
+        """
+        import matplotlib.pyplot as plt
+
+        def to_seconds(timestamp):
+            start_ts = self.start if start is None else start
+
+            if isinstance(timestamp, str):
+                timestamp = to_timestamp(timestamp, start_ts, self.stop)
+
+            return (timestamp - start_ts) / 1e9
+
+        def get_time_range(time_item):
+            if hasattr(time_item, "start") and hasattr(time_item, "stop"):
+                return to_seconds(time_item.start), to_seconds(time_item.stop)
+            elif np.isscalar(time_item) == 1:
+                return to_seconds(time_item), None
+            elif len(time_item) == 2:
+                return to_seconds(time_item[0]), to_seconds(time_item[1])
```

Comment:
Other things than list or tuples have a `len(...)`.

Reply:
I think it's fixed in the new version, but I'm still curious what you were hinting at here specifically?

Reply:
The function seems to be very accepting on terms of what `time_item` can be, and that's by design.
From the code it also seems that anything that the function _does not_ accept should land in the final exception handling branch. But here, there was the possibility that  a `time_item` could slip through and end in this case, which wouldn't handle it properly. Say for instance, a string.

Reply:
Yeah, the string case should be handled now.